game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -1100,6 +1152,12 @@ public void validate(final GameState data) {
             this::resetAtWarPlayers);
       case "atWarCount":
         return MutableProperty.ofReadOnly(this::getAtWarCount);
+      case "haveResources":
```

Comment:
Add annotation` @NoneNls` to parameter `propertyName`.
We want to introduce internationalization and this can help identify the strings that need translation (or as in this case that do not need it).